### DISCUSSION TITLE
Create an extension for ksmtuned?

### DISCUSSION THREAD

**Discussion by MichaelTrip** [siderolabs/talos#9361] [Ideas]
With the release 1.8.0 KSM is now officially enabled in the kernel. Which is good news if you would like to run Virtual machines with Kubevirt for example. 

With most common Linux distributions there is a ksmtuned systemd service that tunes KSM specific needs. I am now thinking about creating an extension with this daemon. But it is a shell script.

 https://github.com/ksmtuned/ksmtuned

What is the best practice in making an extension that contains a shell script?

**Comment by smira** [comment]
I would prefer not to have any shell scripts at all, and probably this could be moved into native Talos code, even though we don't have any spare cycles to work on that.

You're looking towards having a system extension which packages Talos extension service here. An extension service is a rootfs of a container image, so it can be literally anything, e.g. `alpine` + the script, but it should be self-contained within its rootfs.

`hello-world-service` might be a starting point.
